assignments/this.js

```diff
@@ -30,16 +34,57 @@ const dad = {
 dad.greet();
 // Principle 3
 
+//New Binding creates oblects
+
 // code example for New Binding
 
+function CostcoDoorman(greeter) {
+    this.greeting = 'Hello ';
+    this.greeter = greeter;
+    this.speak = function() {
+        console.log(`${this.greeting} ${this.greeter}`);
+    };
+  }
+  
+  const ivan = new CostcoDoorman('Keith');
+  const keith = new CostcoDoorman('Ivan');
+
+  console.log(ivan);
+  console.log(keith);
+
+  keith.speak();
+   ivan.speak();
+
+//   const ivan = {
+//       greeter: 'Ivan',
+//       greeting: 'Hello'
+//       speak: function() {
+//         console.log(`${this.greeting} ${this.greeter}`);
+//       }
+//   }
+  
+
 // Principle 4
 
+//Whenever JavaScript’s call or apply method is used, this is explicitly defined.
+
 // code example for Explicit Binding
-function greet() {
-    console.log(`Hello, my name is ${this.name}`)
-}
-const dad = {
+
+const person = {
     name: 'Tuvok',
     age: 104
 }
-greet.call(dad);
\ No newline at end of file
+const sports = ['baseball', 'football', 'basketball'];
+
+function funTime() {
+    'use strict'
+    console.log(`Hello my name is ${this.name} and I am ${this.age}.  The sports
+    that I like to play are ${sports}`)
+}
+
+    funTime.call(person);
+
+
+
+    const introduceLater = funTime.bind(person);
```

Comment:
I'm impressed with this. `bind` is a very powerful concept in JS that not many developers use often.